content/sensu-go/5.17/guides/secrets-management.md

```diff
@@ -0,0 +1,289 @@
+Then, run:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: VaultProvider
+api_version: secrets/v1
+metadata:
+  name: vault
+spec:
+  client:
+    address: http://localhost:8200
+    token: ROOT_TOKEN
+    version: v2
+    tls: null
+    max_retries: 2
+    timeout: 20s
+    rate_limiter:
+      limit: 10
+      burst: 100
+EOF
+{{< /highlight >}}
+
+### Create your secret
+
+_**NOTE**: Because you aren't using TLS, you will need to add `export VAULT_ADDR=http://127.0.0.1:8200` in your bash profile._
+
+1. Retrieve your PagerDuty service routing key.
+This is the secret you will set up in Vault.
+2. Open a new terminal and run `vault kv put secret/pagerduty key=SERVICE_ROUTING_KEY`.
+Replace `SERVICE_ROUTING_KEY` with your PagerDuty service routing key.
+
+This writes your secret into Vault.
+In this example, the name of the secret is `pagerduty`.
+The `pagerduty` secret contains a key, and you specified that the `key` value is your PagerDuty service routing key.
+
+**Is the service routing key the same as the API Integration key mentioned in PagerDuty docs at https://support.pagerduty.com/docs/generating-api-keys#section-events-api-keys ?**
+
+The `id` value for secrets that target a HashiCorp Vault must start with the name of the secret's path in Vault.
+The Vault dev server is preconfigured with the `secret` keyspace already set up, so we recommend using the `secret/` path for the `id` value while you are learning and getting started with Vault secrets management.
+
+Run `vault kv get secret/pagerduty` to see the secret you just set up.
+
+**STOPPED HERE FRIDAY**
+
+Use sensuctl create to create your secret:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: pagerduty_key
+  namespace: default
+spec:
+  id: secret/pagerduty#key
+  provider: vault
+EOF
+{{< /highlight >}}
+
+Now you can securely pass your PagerDuty service routing key in Sensu checks, handlers, and mutators by referring to the `pagerduty_key` secret.
+In this guide, you'll use your `pagerduty_key` secret in a handler.
+
+## Add a handler
+
+### Register the PagerDuty Handler asset
+
+To begin, register the [Sensu PagerDuty Handler asset][23] with [`sensuctl asset add`][22]:
+
+{{< highlight shell >}}
+sensuctl asset add sensu/sensu-pagerduty-handler:1.2.0 -r pagerduty-handler
+{{< /highlight >}}
+
+This example uses the `-r` (rename) flag to specify a shorter name for the asset: `pagerduty-handler`.
+
+Run `sensuctl asset list --format yaml` to confirm that the asset is ready to use.
+
+**I'm not sure whether any of the content from https://docs.sensu.io/sensu-go/latest/guides/install-check-executables-with-assets/#2-adjust-the-asset-definition needs to be included here.**
+
+With this handler, Sensu can trigger and resolve PagerDuty incidents.
+However, you still need to add your secret to the handler spec so it will require your backend to request secrets from your secrets provider.
```

Comment:
```suggestion
However, you still need to add your secret to the handler spec so that it requires your backend to request secrets from your secrets provider.
```